File: 443-string-compression/string-compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        int write = 0, read = 0;
        int n = chars.size();
        while (read < n) {
            char ch = chars[read];
            int count = 0;
            while (read < n && chars[read] == ch) {
                read++;
                count++;
            }
            chars[write++] = ch;
            if (count > 1) {
                int start = write;
                while (count > 0) {
                    chars[write++] = (count % 10) + '0';
                    count /= 10;
                }
                reverse(chars.begin() + start, chars.begin() + write);
            }
        }
        return write;
    }
};

File: 443-string-compression/string-compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        res=[]
        i=0
        while i<len(chars):
            c=1
            while i+c<len(chars) and chars[i]==chars[i+c]:
                c+=1
            res.append(chars[i])
            if c>1:
                for digit in str(c):
                    res.append(digit)
            i+=c
        chars[:]=res
        return len(chars)